service/controller/resource_set.go

```diff
@@ -145,7 +145,7 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 			return false
 		}
 
-		if key.VersionBundleVersion(cr) == project.NewVersionBundle().Version {
+		if key.OperatorVersion(&cr) == project.Version() {
```

Comment:
I think there is no need for the handles func anymore in case we configure the correct label selector to the controller. Just did it in `aws-operator` as well today. See https://github.com/giantswarm/aws-operator/pull/2361.

Reply:
label selector is the way to go I agree, but I'd like to get us going here. And maybe tackle this in a different PR.

Reply:
With you. I maybe got confused with the PR title.